vault/resource_identity_oidc_role.go

```diff
@@ -0,0 +1,168 @@
+package vault
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/vault/api"
+)
+
+const identityOidcRolePathTemplate = "identity/oidc/role/%s"
+
+func identityOidcRole() *schema.Resource {
+	return &schema.Resource{
+		Create: identityOidcRoleCreate,
+		Update: identityOidcRoleUpdate,
+		Read:   identityOidcRoleRead,
+		Delete: identityOidcRoleDelete,
+		Exists: identityOidcRoleExists,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:        schema.TypeString,
+				Description: "Name of the role.",
+				Required:    true,
+				ForceNew:    true,
+			},
+
+			"key": {
+				Type:        schema.TypeString,
+				Description: "A configured named key, the key must already exist.",
+				Required:    true,
+				ForceNew:    true,
+			},
+
+			"template": {
+				Type:        schema.TypeString,
+				Description: "The template string to use for generating tokens. This may be in string-ified JSON or base64 format.",
+				Optional:    true,
+			},
+
+			"ttl": {
+				Type:        schema.TypeInt,
+				Description: "TTL of the tokens generated against the role in number of seconds.",
+				Optional:    true,
+				Default:     86400,
+			},
+
+			"client_id": {
```

Comment:
I'm curious about why this is added, I don't see it listed [here](https://www.vaultproject.io/api/secret/identity/tokens.html#create-or-update-a-role). I also don't see it listed below in lines 62:64 so I'm thinking it might have slipped in.

Reply:
Please see [comment](https://github.com/terraform-providers/terraform-provider-vault/pull/488#issuecomment-518463146).